plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -166,6 +167,47 @@ public void testSelect()
                         .build());
     }
 
+    @Test(dataProvider = "testSelectWithLimitDataProvider")
+    public void testSelectWithLimit(int numberOfFiles)
+    {
+        // Create table with multiple files
+        assertUpdate("CREATE TABLE test_select_with_limit(k varchar, v integer) WITH (partitioning=ARRAY['truncate(k, 1)'])");
```

Comment:
I see for `numberOfFiles` `10` the partitions: `0` (1 file), `1` (2 files), `2` (2 files), `3` (2 files),`4` (2 files), `5` (1 file). That's not `2 files per partition` as commented below.

Reply:
Thanks, fixing with `10 + i * 5`